v1/status_message.go

```diff
@@ -47,6 +50,17 @@ func (s *StatusMessage) SigPubKey() *ecdsa.PublicKey {
 	return s.TransportLayerSigPubKey
 }
 
+// GenerateUsername returns the 3 words name of the most important signature
+func (s *StatusMessage) GenerateUsername() string {
```

Comment:
Seems unused. I am not convinced `StatusMessage` needs to know about `identicon` and `username` packages. We only do that because we have a public key for each message but username and identicon are properties of contact.

Reply:
We use them in status-go, each message comes with a `Author` with `Name` and `Identicon`, I don't mind though removing the method and have it calculated in `status-go` using `SigPubKey`, similar calculations will have to be done for each client though.

Reply:
We also have these exposed as functions in `messenger.go` so it can be delegated to them. Would that be ok? 

I am trying to limit dependencies between packages as much as possible, especially here to keep `v1` independent. Also, `v1` as a package probably wasn't the best idea...

Reply:
yes, that would work fine, we can just use those to attach he data, will do that